psl/psl-core/src/configuration/configuration_struct.rs

```diff
@@ -26,6 +26,10 @@ impl Configuration {
         }
     }
 
+    pub fn datasource(&self) -> &Datasource {
+        self.datasources.first().unwrap()
+    }
```

Comment:
We can't guarantee that it won't panic, the validation that ensures that there is at least one datasource is optional. Where do we need this method?

Reply:
Everywhere. You see where we pop and unwrap the first datasource in main, now we cannot do that anymore, so this is the referencing equivalent.

Reply:
I think the psl method should return an option. It's ok to unwrap in introspection because we check very early on that there is at least one datasource there

Reply:
https://github.com/prisma/prisma-engines/pull/3264/commits/765ae81b7bab143d1bbca7791d6c73d6b5af42cd